src/services/lock/service/lock.service.ts

```diff
@@ -82,10 +73,6 @@ export class LockService {
       }
 
       return await task(); // Execute the task
-
-      // Optionally, you can decide if the lock should be released immediately after the task
-      // or if it should be held until the next execution cycle.
-      // await this.unregisterLock(lockId);
```

Comment:
Isn't `await this.unregisterLock(lockId);` required before this function returns to unlock the `taskId`?

Reply:
its optional, I foresee fewer problems keeping things on one node until task execution failure and then falling back vs constantly switching but if you disagree I'd appreciate your input